tests/5-UpdateMixedLineEndingsTest.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+use WP_CLI\Tests\TestCase;
+
+class UpdateMixedLineEndingsTest extends TestCase {
+	private static $test_config_path;
+	private static $test_config_lines;
+	private static $config_transformer;
+	public static function set_up_before_class() {
+		self::$test_config_lines = array(
+			"<?php\n",
+			"// this is a demo\r\n",
+			"define( 'DB_NAME', '' );\n",
+			"define( 'DB_HOST', '' );\r\n",
+			"define( 'DB_USER', '' );\n\r",
+			"define( 'DB_COLLATE', '');\n",
```

Comment:
Can we add double blank lines in multiple constellations here as well just to fully demonstrate it works without splitting the combinations?

Reply:
Added them in dd6f0f9, however I think there's another bug that happens when the provided config file has a multiline string containing any of the replaced characters. In theory, we shouldn't be touching them. To fix it properly, we might require a proper PHP parser though, not sure if it already exists somewhere in wp-cli?

Reply:
No, we don't have a PHP Parser around, we tried to avoid adding dependencies to this to make sure it doesn't randomly break from conflicts.

Reply:
> however I think there's another bug that happens when the provided config file has a multiline string containing any of the replaced characters.

Can you open a separate issue for this?